src/InstrumentationEngine.Lib/EnumeratorImpl.h

```diff
@@ -46,13 +46,14 @@ namespace MicrosoftInstrumentationEngine
         {
             HRESULT hr = S_OK;
 
-            if (dwLength != 0)
+            if (dwLength > 0)
```

Comment:
If there are zero elements, then m_members is not initialized correctly or is it already an empty vector without calling Attach?
If the length is negative for some reason, then this method returns S_OK, which I expect it to return E_INVALIDARG

Reply:
m_members would be a nullptr if not iniitialized, however it is only indexed if m_dwLength > 0. With my change, negative lengths would result in m_dwLength = 0 which is would still be valid.

---
In reply to: [475785866](https://github.com/microsoft/CLRInstrumentationEngine/pull/285#discussion_r475785866) [](ancestors = 475785866)

Reply:
I could return E_INVALIDARG, I'll double check the callers.

---
In reply to: [475805050](https://github.com/microsoft/CLRInstrumentationEngine/pull/285#discussion_r475805050) [](ancestors = 475805050,475785866)